drivers/place/booking_check_in_helper.cr

```diff
@@ -0,0 +1,279 @@
+    no_show_url:  "https://domain.com/meeting/no-show",
+
+    jwt_private_key: <<-STRING
+-----BEGIN RSA PRIVATE KEY-----
+MIIEpAIBAAKCAQEAt01C9NBQrA6Y7wyIZtsyur191SwSL3MjR58RIjZ5SEbSyzMG
+3r9v12qka4UtpB2FmON2vwn0fl/7i3Jgh1Xth/s+TqgYXMebdd123wodrbex5pi3
+Q7PbQFT6hhNpnsjBh9SubTf+IeTIFeXUyqtqcDBmEoT5GxU6O+Wuch2GtbfEAmaD
+roy+uyB7P5DxpKLEx8nlVYgpx5g2mx2LufHvykVnx4bFzLezU93SIEW6yjPwUmv9
+R+wDM/AOg60dIf3hCh1DO+h22aKT8D8ysuFodpLTKCToI/AbK4IYOOgyGHZ7xizX
+HYXZdsqX5/zBFXu/NOVrSd/QBYYuCxbqe6tz4wIDAQABAoIBAQCEIRxXrmXIcMlK
+36TfR7h8paUz6Y2+SGew8/d8yvmH4Q2HzeNw41vyUvvsSVbKC0HHIIfzU3C7O+Lt
+9OeiBo2vTKrwNflBv9zPDHHoerlEBLsnNwQ7uEUeTWM9DHdBLwNaLzQApLD6q5iT
+OFW4NfIGpsydIt8R565PiNPDjIcTKwhbVdlsSbI87cLkQ9UuYIMRkvXSD1Q2cg3I
+VsC0SpE4zmfTe7YTZQ5yTxtsoLKPBXrSxhhGuhdayeN7A4YHFYVD39RuQ6/T2w2a
+W/0UaGOk8XWgydDpD5w9wiBdH2I4i6D35IynCcodc5JvmTajzJT+xj6aGjjvMSyq
+q5ZdwJ4JAoGBAOPdZgjbOCf3ONUoiZ5Qw/a4b4xJgMokgqZ5QGBF5GqV1Xsphmk1
+apYmgC7fmab/EOdycrQMS0am2FmtwX1f7gYgJoyWtK4TVkUc5rf+aoWi0ieIsegv
+rjhuiIAc12+vVIbegRgnq8mOI5icrwm6OkwdqHkwTt6VRYdJGEmu67n/AoGBAM3v
+RAd5uIjVwVDLXqaOpvF3pxWfl+cf6PJtAE5y+nbabeTmrw//fJMank3o7qCXkFZR
+F0OJ2tmENwV+LPM8Gy3So8YP2nkOz4bryaGrxQ4eMA+K9+RiACVaKv+tNx/NbyMS
+e9gg504u0cwa60XjM5KUKrmT3RXpY4YIfUPZ1J4dAoGAB6jalDOiSJ2j2G57acn3
+PGTowwN5g9IEXko3IsVWr0qIGZLExOaZxaBXsLutc5KhY9ZSCsFbCm3zWdhgZ7GA
+083i3dj3C970iHA3RToVJJbbj56ltFNd/OGiTwQpLcTsB3iVSFWVDbpsceXacG5F
+JWfd0O0RyaOk6a5IVbm+jMsCgYBglxAOfY4LSE8y6SCM+K3e5iNNZhymgHYPdwbE
+xPMrWgpfab/Evi2dBcgofM+oLU663bAOspMeoP/5qJPGxnNtC7ZbSMZNL6AxBVj+
+ZoW3uHsMXz8kNL8ixecTIxiO5xlwltPVrKExL46hsCKYFhfzcWGUx4DULTLMBCFU
++M/cFQKBgQC+Ite962yJOnE+bjtSReOrvR9+I+YNGqt7vyRa2nGFxL7ZNIqHss5T
+VjaMgjzVJqqYozNT/74pE/b9UjYyMzO/EhrjUmcwriMMan/vTbYoBMYWvGoy536r
+4n455vizig2c4/sxU5yu9AF9Dv+qNsGCx2e9uUOTDUlHM9NXwxU9rQ==
+-----END RSA PRIVATE KEY-----
+STRING
+  })
+
+  def on_load
+    on_update
+  end
+
+  # See: https://crystal-lang.org/api/latest/Time/Format.html
+  @date_time_format : String = "%c"
+  @time_format : String = "%l:%M%p"
+  @date_format : String = "%A, %-d %B"
+  @timezone : Time::Location = Time::Location::UTC
+
+  protected getter! prompt_after : Time::Span
+  protected getter! present_from : Time::Span
+
+  getter? meeting_pending : Bool = false
+  getter? people_present : Bool = false
+  getter current_meeting : PlaceCalendar::Event? = nil
+
+  # This last meeting id we prompted
+  @prompted : String = ""
+
+  # The URLS we want to send to the user
+  @check_in_url : String = ""
+  @no_show_url : String = ""
+  @domain : String = ""
+
+  @jwt_private_key : String = ""
+
+  def on_update
+    @jwt_private_key = setting(String, :jwt_private_key)
+
+    @prompt_after = (setting?(Int32, :prompt_after) || 10).minutes
+    @present_from = (setting?(Int32, :present_from) || 5).minutes
+
+    @check_in_url = setting(String, :check_in_url)
+    @no_show_url = setting(String, :no_show_url)
+    @domain = URI.parse(@check_in_url).host.not_nil!
+
+    subscriptions.clear
+    system.load_complete do
```

Comment:
Does this need to be in a load_complete callback? I though subscriptions support late binding?

Reply:
you are correct